src/flag_gems/utils/pointwise_dynamic.py

```diff
@@ -355,23 +355,28 @@ def generate_destination_passing_pointwise_wrapper(
     wrapper_signature: str = f"def {wrapper_name}({parameters}):"
     code.writeline(wrapper_signature)
 
-    # task partitioning, 1d task indexing
-    tile_size = 512
-    num_warps = 4
-    if rank == 0:  # special case with rank-0, only 1 element to compute
-        tile_size = 32
-        num_warps = 1
-
     with code.indent():
         # docstring
         wrapper_docstring = docstring_for_destination_passing_wrapper(op_desc)
         code.writeline(wrapper_docstring)
 
-        shapes_str = ", ".join(
-            f"in{i}.shape" for i in range(op_desc.num_input_tensors())
-        )
-        code.writeline(f"shape = broadcast_shapes([{shapes_str}])")
+        code.writeline("shape = out0.shape")
         code.writeline("num_tasks = volume(shape)")
+
+        if rank > 0:
+            code.writeline("tile_size = min(8192, triton.next_power_of_2(num_tasks))")
+            code.writeline(
+                "num_warps = 4 if tile_size <= 1024 else (8 if tile_size <= 4096 else 16)"
+            )
+            code.writeline("num_ctas = min(65535, triton.cdiv(num_tasks, tile_size))")
```

Comment:
will it limit the grid size on NV gpu?

Reply:
Yes，until we find a way to read the max grid size at code generation time.
But maybe it would not reduce the performance since 65535 is already larger than the number of SMs. So it can saturate the device.

Reply:
Can we got the max grid size with different device？